### DISCUSSION TITLE
Not able to insert a record (JS pb_hooks)

### DISCUSSION THREAD

**Discussion by sz-dev** [pocketbase/pocketbase#4559] [Q&A] [answered]
<img width="944" alt="issue2" src="https://github.com/pocketbase/pocketbase/assets/66676736/cbbdf54a-cf73-43b7-acaa-3bc7bbd00607">
<img width="864" alt="issue1" src="https://github.com/pocketbase/pocketbase/assets/66676736/8d1f2b64-a0e3-4785-b240-8cde7fb1dcd0">

The event is working, but not able to insert to another collection.
Tried so many variations :(
even tried following and didn't work
![image](https://github.com/pocketbase/pocketbase/assets/66676736/9af02a0f-231c-4610-8af5-1d1b0367c12f)
also tried inserting to a simple collection with only static text, didn't work

**Comment by ganigeorgiev** [comment]
Please don't post code as screenshot. It is hard to read and there is no easy way to copy it and tried locally (_you can paste formatted codeblock by wrapping it with 3 backticks ` ``` `_).

If you are still not able to resolve it, please provide clear reproducible steps or at least elaborate on your use case and what "didn't work" means (is there any error in the console or in the _Admin UI > Logs_ section?).

**Reply by ganigeorgiev** [reply, marked as answer]
Also note that there is no such method as `createRecord`, maybe you wanted `new Record(collection)`?

Please also make sure to check https://pocketbase.io/docs/js-records/#create-new-record.

**Reply by sz-dev** [reply]
Hey

Thank you so much, that was the problem unfortunately, syntax error :(
I love the Java extension feature!

And next time, I will post code snippet instead of screenshots.. I am new to github discussions.

my collection names are 'Insurance' & 'log' - - below is working perfectly.
``
--------------------------------------
onModelAfterCreate(async (e) => {
    if (e.model.tableName() === "Insurance") {
        const insuranceRecord = e.model;
        const clinicRef = insuranceRecord.get("ClinicsRef");
        const insuranceRef = insuranceRecord.get("Insurance");

        console.log("New Insurance record added:", JSON.stringify(insuranceRecord));
        console.log("ClinicRef:", clinicRef);
        console.log("InsuranceRef:", insuranceRef);      
		
		  // Save the new record
		try {
			const collection = await $app.dao().findCollectionByNameOrId("log")
		   const record = new Record(collection, {
			// bulk load the record data during initialization
			"name": "Lorem ipsum"
			});
			await $app.dao().saveRecord(record);
			console.log("evStage record saved.");
			} 
	catch (error) {
				console.error("Error saving evStage record:", error);
			}
        console.log("evStage record saved requested.");
		
    }
}, "Insurance");
-------------------------------------------
